checks/cs50/2017/x/scratch/checks.py

```diff
@@ -0,0 +1,139 @@
+import json
+import os
+import sys
+
+sys.path.append(os.getcwd())
+from check50 import *
+
+class Scratch(Checks):
+    
+    @check()
+    def valid(self):
+        """project exists and is valid Scratch program"""
+        self.exists("project.sb2")
+
+        # Ensure that unzipped .sb2 file contains .json file
+        self.spawn("unzip project.sb2").exit()
+        self.exists("project.json")
+    
+    @check("valid")
+    def two_sprites(self):
+        """project contains at least two sprites"""
+        project = json.loads(File("project.json").read())
+
+        # Loop over children: includes sprites and stages.
+        num_sprites = sum("costumes" in child for child in project["children"])
+
+        if num_sprites < 2:
+            raise Error("Only {} sprite{} found, 2 required.".format(num_sprites,
+                "" if num_sprites == 1 else "s"))
+
+    @check("valid")
+    def non_cat(self):
+        """project contains a non-cat sprite"""
+        project = json.loads(File("project.json").read())
+        non_cat = False
+    
+        # As a heuristic, only the cat has a "meow" sound.
+        for child in project["children"]:
+        
+            # Skip over any non-sprites (e.g. backdrops).
+            if "costumes" not in child:
+                continue
+            
+            # If sprite has no sounds, it's probably not a cat.
+            if "sounds" not in child:
+                return
+
+            can_meow = any(sound["soundName"] == "meow" for sound in child.get("sounds", []))
+
+            # If it doesn't meow, we've found a non-cat sprite.
+            if not can_meow:
+                return
+        
+        # If we haven't returned, then no non-cat sprite found.
```

Comment:
This might be a reasonable place to use a `for/else` (whatever you think is clearer)

i.e.

```python
for child in project["children"]:

   # Skip over any non-sprites (e.g. backdrops).
    if "costumes" not in child:
        continue

   # If it doesn't meow, we've found a non-cat sprite
    if not any(sound["soundName"] == "meow" for sound in child.get("sounds, [])):
        break
else:
    raise Error("Requires a non-cat sprite")